src/flag_gems/utils/type_utils.py

```diff
@@ -0,0 +1,9 @@
+import torch._prims_common as utils
+
+
+def type_promotion(*args, type_promotion: utils.ELEMENTWISE_TYPE_PROMOTION_KIND):
```

Comment:
Maybe we can use a simpler implementation for type promotion rules.

Reply:
Yeah, we can simplify the logic!  but I have two concerns. 1. Will Gems support dynamic-shape with maybe `sympy` in the feature？2. I‘m going to figure out  if there are differences among `torch.promote_types`  `torch.result_type` and  ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT

Reply:
We may not support symbolic shape since flaggems is basically designed for processing real tensors, with concrete shapes.